tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,266 @@
+            self.assertEqual(response.data["system_defined"], False)
+
+    def test_update_taxonomy_permission_anon(self):
+        url = reverse("taxonomy-detail", kwargs={"id": self.taxonomy.id})
+
+        response = self.client.put(url, {"name": "new name"})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_create_taxonomy_permission_user(self):
+        url = reverse("taxonomy-detail", kwargs={"id": self.taxonomy.id})
+
+        self.client.force_authenticate(user=self.user)
+        response = self.client.put(url, {"name": "new name"})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    # def test_update_taxonomy_system_defined(self): #ToDo: Remove?
+    #     url = reverse("taxonomy-detail", kwargs={"id": self.taxonomy_system.id})
+
+    #     with patch.object(
+    #         self.user, "has_perm", _has_perm_change_taxonomy
+    #     ):
+    #         self.client.force_authenticate(user=self.user)
+    #         response = self.client.put(url, {"name": "new name"})
+    #         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_update_taxonomy(self):
+        # Happy path
+        taxonomy = Taxonomy.objects.create(name="test update taxonomy")
+        taxonomy.save()
+
+        url = reverse("taxonomy-detail", kwargs={"id": taxonomy.id})
+        with patch.object(
+            self.user, "has_perm", _has_perm_change_taxonomy
+        ):
+            self.client.force_authenticate(user=self.user)
+            response = self.client.put(url, {"name": "new name"})
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            taxonomy = Taxonomy.objects.get(id=taxonomy.id)
+            self.assertEqual(taxonomy.name, "new name")
+
+    def test_delete_taxonomy_permission_anon(self):
+        url = reverse("taxonomy-detail", kwargs={"id": self.taxonomy.id})
+
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_taxonomy_permission_user(self):
+        url = reverse("taxonomy-detail", kwargs={"id": self.taxonomy.id})
+
+        self.client.force_authenticate(user=self.user)
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_delete_taxonomy(self):
+        # Happy path
+
+        taxonomy = Taxonomy.objects.create(name="test delete taxonomy")
+        taxonomy.save()
+
+        url = reverse("taxonomy-detail", kwargs={"id": taxonomy.id})
+        with patch.object(
+            self.user, "has_perm", _has_perm_delete_taxonomy
+        ):
+            self.client.force_authenticate(user=self.user)
+            response = self.client.delete(url)
+            self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+            with self.assertRaises(Taxonomy.DoesNotExist):
+                Taxonomy.objects.get(id=taxonomy.id)
```

Comment:
Please consolidate your tests wherever possible using ddt, 

e.g.

```suggestion
    @ddt.data(
        (None, status.HTTP_403_FORBIDDEN),
        ('user', status.HTTP_403_FORBIDDEN),
        ('staff', status.HTTP_204_NO_CONTENT),
    )
    @ddt.unpack
    def test_delete_taxonomy(self, user_attr, expected_status):
        url = reverse("oel_tagging:taxonomy-detail", kwargs={"id": self.taxonomy.id})

        user = getattr(self, user_attr) if user_attr else None
        if user:
            self.client.force_authenticate(user=user)
        response = self.client.delete(url)
        self.assertEqual(response.status_code, expected_status)

        # If we were able to delete the taxonomy, check that it's really gone
        if status.is_success(expected_status):
            response = self.client.get(url)
            self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Reply:
Note that the last line of my suggested change fails, but it shouldn't -- the `detail` view should return a 404 when the taxonomy doesn't exist.